### PR TITLE
Add support for ZCU106

### PR DESCRIPTION
**Pull request by christian-lanius**
This pull request adds support for the ZCU106 eval board from Xilinx. It is basically identically to the ZCU104, upon which this is based.
I have basically copied all the files from the ZCU104 folder and adapted them (mostly changing the constraints to fix the pinout) where necessary. 
I am unsure if this approach is the correct one. The one with less duplicate files would be to reuse the "rev" approach used for the Nexys boards. In that case, there would be one ZCU target with revisions 104 and 106. If this is requested, I will modify the code appropriately.

I have verified that openocd with gdb works to connect to the SoC and UART can be used to write out to the console.
One thing to pay attention to, is that the pmod headers are switched: The pmod0 header is hooked up to the pmod1 ports and the other way around. This makes it easier to attach the JTAG debugger, as pmod 1 is male and pmod0 is female.

### CODE REVIEW

**Comment by meggiman** [comment]
Hello @christian-lanius 
Thanks a lot for this awesome contribution. I'm running into some issues when I try to run your port in Vivado myself. Which version of Vivado where you using when you successfully created the bitstream using the make zcu106 target?

**Comment by christian-lanius** [comment]
Hi @meggiman,
I have run the generation with Vivado 2019.2 on Ubuntu 20.04 LTS.
What kind of issue are you encountering?

Kind regards
Christian

File: fpga/pulpissimo-zcu106/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# PULPissimo on the Xilinx ZCU104 Board
```
Comment (meggiman) [inline fpga/pulpissimo-zcu106/README.md:1]: Please update this README file to be inline with the port (i.e. update default frequencies, titles etc.)

File: fpga/pulpissimo-zcu106/elf_run.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+trap "exit" INT TERM
+trap "kill 0" EXIT
+
+
+SCRIPTDIR=$(dirname $0)
+UART_TTY=${PULP_ZCU104_UART_TTY:=/dev/ttyUSB0}
+UART_BAUDRATE=${PULP_ZCU104_UART_BAUDRATE:=115200}
+
+#Execute gdb and connect to openocd via pipe
+$OPENOCD/bin/openocd -f $SCRIPTDIR/openocd-zcu104.cfg &
```
Comment (meggiman) [inline fpga/pulpissimo-zcu106/elf_run.sh:12]: Please make sure to replace all zcu104 remain with the zcu106 equivalents. I guess a global string replace in the pulpissimo-zcu106 folder should do the trick

**Review by meggiman — CHANGES_REQUESTED** [review]
Except for a few minor things this PR looks good to me. Great work!

**Comment by meggiman** [comment]
> Hi @meggiman, I have run the generation with Vivado 2019.2 on Ubuntu 20.04 LTS. What kind of issue are you encountering?
> 
> Kind regards Christian

While trying your PR I had some elaboration issues with the latest version of Vivado available on my machine (2020.2). The issue is unrelated to your PR. I will have to open a separate issue for that. For now

**Pull request merged**